packages/std/src/binary.rs

```diff
@@ -15,17 +16,26 @@ use crate::errors::{StdError, StdResult};
 pub struct Binary(#[schemars(with = "String")] pub Vec<u8>);
 
 impl Binary {
+    /// Base64 encoding engine used in conversion to/from base64.
+    const B64_ENGINE: GeneralPurpose = GeneralPurpose::new(
+        &base64::alphabet::STANDARD,
+        base64::engine::GeneralPurposeConfig::new()
+            .with_decode_padding_mode(base64::engine::DecodePaddingMode::Indifferent),
+    );
```

Comment:
Why configure a custom engine here instead of using the standard one?

In any case, we should test the expected behaviour here. There are from_base64 teste here but not nicely organized by function yet: https://github.com/CosmWasm/cosmwasm/blob/0e483094f96b39e454d89a0e9913cb3434cf804b/packages/std/src/binary.rs#L314-L339

Reply:
> Why configure a custom engine here instead of using the standard one?

Because the two available in base64 are STANDARD which require padding when decoding and NO_PAD which requires no padding when decoding and based on tests we want one which encodes with padding but decodes string with or without padding.

Reply:
> which require padding when decoding

okay but why is this a problem. Isn't that the current behaviour anyways?

Reply:
No, the current behaviour (as tested in `test_shortened_string`) is to accept padding or no padding when decoding.

Reply:
okay, nevermind, it's already there